Source/WebCore/dom/Position.cpp

```diff
@@ -342,9 +344,6 @@ Position Position::previous(PositionMoveType moveType) const
     if (!node)
         return *this;
 
-    // FIXME: Negative offsets shouldn't be allowed. We should catch this earlier.
-    ASSERT(deprecatedEditingOffset() >= 0);
```

Comment:
Actually, this function still returns int so we should be asserting that it's non-negative.

Reply:
So I should close this PR, since we need these assertions still? Or we can add Assert at top in L146 and remove these?

Reply:
No, I don't think we should remove these assertions or add that assertion.
Instead, the rest of code changes to use `!m_offset` instead of `m_offset > 0` is good.
We need to rename the PR / bug title accordingly though.

Reply:
Done.